src/main/java/com/limechain/sync/warpsync/SyncedState.java

```diff
@@ -103,7 +103,7 @@ public NeighbourMessage getNeighbourMessage() {
         );
     }
 
-    public void syncCommit(CommitMessage commitMessage, PeerId peerId) {
+    public synchronized void syncCommit(CommitMessage commitMessage, PeerId peerId) {
```

Comment:
Can we add a comment about the race condition? In the future someone might wonder why this part is synchronized